typescript/introduction/tsc-interfaces-classes/tsc-interfaces-intro.md

```diff
@@ -0,0 +1,113 @@
+---
+author: kapnobatai136
+
+type: normal
+
+category: must-know
+
+aspects:
+  - introduction
+
+---
+
+# Introduction to Interfaces
+
+---
+## Content
+
+In TypeScript, the shape that represents a collection of key-value pairs is an object.
+
+One way to declare a type of an object is with `{}`:
+
+```ts
+function f(companyObj: { name: string }) {
+  console.log(companyObj.name);
+}
+const someObject = {
+  name: "Enki"
+};
+f(someObject);
+```
+
+It won't care about the order of the properties or if any additional ones exist.
+
+If we wanted to reuse the type of `companyObj` at multiple places in our program, we can extract it into an interface:
```

Comment:
not the only way to do it - there are also type aliases: https://pawelgrzybek.com/typescript-interface-vs-type/
which allow for the computation of their signature

Reply:
https://www.typescriptlang.org/docs/handbook/advanced-types.html#interfaces-vs-type-aliases
more info on this

let's maybe add an insight on this as well?

Reply:
I think that should be moved into the next course right? Not needed for the intro one?

Reply:
why? type aliases and interfaces are very similar and almost used interchangeably - why should we, in the intro, say that interfaces are the only way to do it

Reply:
I don't think we say that it is **the only way to do it**, but rather it is **one of the ways to do it**